### ISSUE TITLE
The key builder of an oracle result cannot be put a question to, so an input dropped from it turns no case red

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#561]
### What minimal example or steps are needed to reproduce the bug?

`scripts/oracles/compare.ts` builds the key of every oracle result in a function of its own:

```ts
function inputsOf (oracle: string, revision: string): Record<string, string> {
	return { oracle, lib: hashAt(revision, `lib`), oracles: hashSourcesAt(`worktree`, `scripts/oracles`), harness: hashSourcesAt(`worktree`, `scripts/harness`), lock: hashAt(`worktree`, `pnpm-lock.yaml`) }
}
```

Take `oracles` out of that object, or take it at `revision` rather than at the working tree, and run `make verify`. It passes. The file cannot be imported by a case — it reads `argv`, runs all six oracles and writes a report as it loads — so no case calls that function, and nothing else asks anything about it either.

### What did you expect to happen?

That an input dropped from the key of an oracle turns a case red, as one dropped from the key of a sweep does since #553. That issue moved the sweeps' key builder into `scripts/sweeps/key.ts`, a module of its own, and `scripts/sweeps/key.test.ts` pins it: five cases, red on thirteen mutations, among them each input taken at the side rather than at the working tree.

### What actually happened?

Nothing pins the oracles' side. The inputs there are right today — that is what #553 checked while it was measuring its own — so this is a latent defect rather than a wrong answer being handed out: the day one of them is dropped, six oracles read a kept result back over a question that has moved, and the run reports that nothing moved.

### Notes

The commit message of #544 called this one a mutation no suite could catch — "either caller left on `hashAt` — the suite cannot catch, since both are scripts that run on import". That went a step too far, and #553 showed where: what cannot be done is calling such a file, not asking about it. Its fifth case reads `scripts/sweeps/run.ts` as text and asks that it name `inputsOf` and no hash of its own, which is how `lib/syntaxes/index.test.ts` holds a rule about modules it does not run either.

#553 closed one of the two callers and left this one, since a branch is reviewed against one axis and its axis was the sweeps' key.

### Do you have a proposal to fix the bug?

The shape #553 used, both halves of it. `inputsOf` into `scripts/oracles/key.ts` with a `key.test.ts` beside it, and a case reading `compare.ts` as text to say that the key it builds is that function's answer. The fourth case of the sweeps' suite is the one worth copying: it fabricates a side with `git mktree` whose every path the key hashes holds something the repository does not, so that no input can pass the case by standing the same on both sides, and it needs no history, which a clone of depth 1 does not have.

Nothing about what the key carries changes: `compare.ts` already names `scripts/oracles`, `scripts/harness`, `lib/` and the lock file. Moving the builder moves no key, so no result in the store stops answering.
